### PR TITLE
Added daemon sets / resiliency support

### PR DESCRIPTION
**Pull request by davidbalazs93**

### CODE REVIEW

**Comment by davidbalazs93** [comment]
Also upgraded k8s client from 1.5 to 2.0

**Comment by davidbalazs93** [comment]
at this moment I'm using the default configuration of circle CI, but when we will manage vendoring with govendor and will introduce coveralls, I will add the custom circle CI config

File: controller_test.go [hunk 1]
```diff
@@ -38,11 +38,11 @@ func (m *MockService) getServicesByNames(serviceNames []string) []service {
```
Comment (MihaiSorinMoisa) [inline controller_test.go]: There are still flows that are not covered by the tests.

File: checkerService.go [hunk 2]
```diff
@@ -20,24 +19,64 @@ type healthcheckResponse struct {
 	}
 }
 
-func (hs *k8sHealthcheckService) checkServiceHealth(serviceName string) (string, error) {
-	k8sDeployments, err := hs.k8sClient.Extensions().Deployments("default").List(api.ListOptions{LabelSelector: labels.SelectorFromSet(labels.Set{"app": serviceName})})
+func (hs *k8sHealthcheckService) checkServiceHealth(service service) (string, error) {
+	var noOfAvailablePods, noOfUnavailablePods int32
+	var err error
+	if service.isDaemon {
+		noOfAvailablePods, noOfUnavailablePods, err = hs.checkServiceHealthForDaemonset(service)
```
Comment (tamas-molnar) [inline checkerService.go]: Maybe a better name would be `getPodAvailabilityForDaemonSet/Deployment`?

Comment (davidbalazs93) [inline checkerService.go]: yep, I've changed it

**Comment by khatton-ft** [comment]
Too late for this PR, but haven't we agreed to use `govendor` and _not_ to check in all the third party packages in `/vendor`?

**Comment by davidbalazs93** [comment]
vendoring with govendor was added in this PR: https://github.com/Financial-Times/upp-aggregate-healthcheck/pull/5

**Pull request merged**